core/utils.py

```diff
@@ -91,6 +96,27 @@ class ExplorationConversionError(Exception):
     pass
 
 
+def open_file(
+        filename: str, mode: Union[TextModeTypes, BinaryModeTypes],
+        encoding: str='utf-8', newline: Union[str, None]=None) -> IO[Any]:
```

Comment:
Can you use `@overload` here to specify either `IO[str]` or `IO[bytes]`? You can use the ModeTypes for this.

Reply:
yes i did implement this but a problem that came across is that for some reason there occurs a (no overload found) error for  open_file function that contains some argument types not defined in the original functions ( ex https://github.com/oppia/oppia/blob/f6bfbcfbe301a6a17409da00cd45e953358cf694/core/domain/topic_domain_test.py#L760 ) . In this example the encoding is None which is not an encoding type for open_file . I was unable to remove this mypy error , however i'll try it again and keep you posted about it .

Reply:
okey , done !! PTAL @vojtechjelinek